### PR TITLE
Fix log files and remove env_logger

### LINKED ISSUE #475: Flag `--log` is broken?
I found that the current youki can't write logs to a file.

```
$ sudo ./youki --log /tmp/youki.log create -b ~/workspace/tmp/mycontainer mycontainer

# file is created, but empty
$ ls -l /tmp/youki.log
-rw-r--r-- 1 root root 0 Nov 14 12:45 /tmp/youki.log
```

youki uses env_logger, but `env_logger::Target::Pipe` is broken according to https://github.com/env-logger-rs/env_logger/issues/208.

### PR DESCRIPTION
**Pull request by yihuaf**
Fix #475 

As mentioned in #475, the log file is broken due to env_logger breaks. Originally we were hoping using env_logger can save us from maintaining our own logger, but it seems that env_logger is not a good fit. While there is an issue filed with env_logger to fix file logging target, it is unlikely to be fixed in a reasonable time frame.

For logger, we require:
1. Log to file
2. Ability to set formatter for JSON
3. Not complex to configure and lightweight since our use case is simple and we would like to maintain performance.

For now, let's fall back to use our own logger implementation. It is simple enough and fits our use case. I am surprised that there is no good out of box logger for Rust similar to `logrus` for Golang. Most of the logging libraries in Rust either lacks features or is not well maintained. https://github.com/tokio-rs/tracing seems to be a promising substitute for `log` in rust, but it requires more work.

### CODE REVIEW

File: crates/youki/src/logger.rs [hunk 1]
```diff
@@ -66,47 +74,94 @@ fn detect_log_level(is_debug: bool) -> Result<LevelFilter> {
...
-    write!(f, "{{")?;
-    write!(f, "\"level\":\"{}\",", record.level())?;
-    write!(f, "\"time\":\"{}\"", chrono::Local::now().to_rfc3339(),)?;
-    write!(f, ",\"msg\":")?;
-    // Use serde_json here so we don't have to worry about escaping special characters in the string.
-    serde_json::to_writer(f.by_ref(), &record.args().to_string())?;
-    writeln!(f, "}}")?;
+struct YoukiLogger {
+    /// Indicates level up to which logs are to be printed
+    level: Option<log::Level>,
```
Comment (utam0k) [inline crates/youki/src/logger.rs:79]: What is the case of `None`?

Comment (Furisto) [reply crates/youki/src/logger.rs:79]: None means logging is disabled. None is returned by to_level if a logging level of 'Off' is specified.

**Review by Furisto — APPROVED** [review]

**Comment by Furisto** [comment]
Looks good!

File: crates/youki/src/logger.rs (hunk 1 above)
Comment (utam0k) [reply crates/youki/src/logger.rs:79]: I forgot about it! You're right.

**Pull request merged**